internal/pkg/vcsutil/vcs_util.go

```diff
@@ -0,0 +1,276 @@
+package vcsutil
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/hashicorp/go-hclog"
+	"github.com/pkg/errors"
+)
+
+var (
+	sshRemoteRegexp  *regexp.Regexp
+	httpRemoteRegexp *regexp.Regexp
+)
+
+func init() {
+	// Regex matching http/https remotes, tokenizing the unique components for replacement.
+	sshRemoteRegexp, _ = regexp.Compile(`git@(.*?):(.*\.git)`)                 // Example: git@git.test:testorg/testrepo.git
+	httpRemoteRegexp, _ = regexp.Compile(`http[s]?:\/\/(.*?\..*?)\/(.*\.git)`) // Example: https://git.test/testorg/testrepo.git
+}
+
+type VcsChecker struct {
+	log hclog.Logger
+
+	// path is the local path to the cloned git repo
+	path string
+
+	// remoteName is the name of the remote that corresponds to the url
+	remoteName string
+
+	// internal go-git repo
+	repo *git.Repository
+}
+
+// NewVcsChecker creates a new configured VcsChecker.
+// NOTE: VcsChecker subprocceses to the `git` command. Git must be installed.
+func NewVcsChecker(log hclog.Logger, path string, remoteUrl string) (*VcsChecker, error) {
```

Comment:
`New` alone here, probably.

Reply:
I just turned all the methods into functions. We weren't tracking anything stateful - we were just saving the logger and path for convenience, but I think it might have been hurting readability.